common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java

```diff
@@ -333,33 +333,63 @@ protected Path getRecoveryPath(String fileName) {
   }
 
   /**
+   * Check the chosen DB file available or not.
+   */
+  protected Boolean checkFileAvailable(File file) {
+      if (file.canWrite()){
+        return true;
+      }
+
+      return false;
+  }
+
+  /**
    * Figure out the recovery path and handle moving the DB if YARN NM recovery gets enabled
    * when it previously was not. If YARN NM recovery is enabled it uses that path, otherwise
    * it will uses a YARN local dir.
    */
   protected File initRecoveryDb(String dbName) {
+    Boolean bolRecoveryPathAvailable = true;
+
     if (_recoveryPath != null) {
         File recoveryFile = new File(_recoveryPath.toUri().getPath(), dbName);
-        if (recoveryFile.exists()) {
+
+        bolRecoveryPathAvailable = checkFileAvailable(recoveryFile);
+        logger.info("Recovery path {} ldb available: {}.", _recoveryPath, bolRecoveryPathAvailable);
+        if (recoveryFile.exists() && bolRecoveryPathAvailable) {
           return recoveryFile;
         }
     }
+
+    // If recovery path unavailable, no use it any more.
+    if (!bolRecoveryPathAvailable) {
```

Comment:
I think recovery path is set by user or use yarn default, user should make sure the availability of this directory, and yarn internally relies on it. It doesn't make sense to change to another disk if recovery path is unavailable.